source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -33,6 +39,53 @@ class ComponentInterface : NodeT {
   );
 
 protected:
+  /**
+   * @brief Add a parameter.
+   * @details This method stores a pointer reference to an existing Parameter object in the local parameter map
+   * and declares the equivalent ROS parameter on the ROS interface.
+   * @param parameter A ParameterInterface pointer to a Parameter instance.
+   */
+  void add_parameter(const std::shared_ptr<state_representation::ParameterInterface>& parameter);
+
+  /**
+   * @brief Add a parameter.
+   * @details This method creates a new Parameter object instance to reference in the local parameter map
+   * and declares the equivalent ROS parameter on the ROS interface.
+   * @tparam T The type of the parameter
+   * @param name The name of the parameter
+   * @param value The value of the parameter
+   */
+  template<typename T>
+  void add_parameter(const std::string& name, const T& value);
+
+  /**
+   * @brief Get a parameter by name.
+   * @param name The name of the parameter
+   * @return The ParameterInterface pointer to a Parameter instance
+   */
+  std::shared_ptr<state_representation::ParameterInterface> get_parameter(const std::string& name);
```

Comment:
Shouldn't those getters now be marked `const` as well or is this a different situation now?